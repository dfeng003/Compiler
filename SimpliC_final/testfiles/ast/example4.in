int main() {
int a;
int b;
a = read();
b = read();
print(gcd1(a, b));
print(gcd2(a, b));
int diff = gcd1(a, b) - gcd2(a, b);
print(diff);
return 0;
}