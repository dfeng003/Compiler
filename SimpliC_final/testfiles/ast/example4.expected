Program
  List
    Func_del
      IdFun ID="main"
      List
      List
        Var_del
          IdVar ID="a"
          Opt
        Var_del
          IdVar ID="b"
          Opt
        Asst
          IdUse ID="a"
          Func_call
            IdCall ID="read"
            List
        Asst
          IdUse ID="b"
          Func_call
            IdCall ID="read"
            List
        Func_call_stmt
          IdCall ID="print"
          List
            Func_call
              IdCall ID="gcd1"
              List
                IdUse ID="a"
                IdUse ID="b"
        Func_call_stmt
          IdCall ID="print"
          List
            Func_call
              IdCall ID="gcd2"
              List
                IdUse ID="a"
                IdUse ID="b"
        Var_del
          IdVar ID="diff"
          Opt
            Minus
              Func_call
                IdCall ID="gcd1"
                List
                  IdUse ID="a"
                  IdUse ID="b"
              Func_call
                IdCall ID="gcd2"
                List
                  IdUse ID="a"
                  IdUse ID="b"
        Func_call_stmt
          IdCall ID="print"
          List
            IdUse ID="diff"
        Return
          Numeral NUMERAL="0"
